src/platform_impl/linux/x11/window.rs

```diff
@@ -1702,7 +1709,28 @@ impl UnownedWindow {
 
     #[inline]
     pub fn request_activation_token(&self) -> Result<AsyncRequestSerial, NotSupportedError> {
-        Err(NotSupportedError::new())
+        // Get the title from the WM_NAME property.
+        let atoms = self.xconn.atoms();
+        let title = {
+            let title_bytes = self
+                .xconn
+                .get_property(self.xwindow, atoms[_NET_WM_NAME], atoms[UTF8_STRING])
+                .expect("Failed to get WM_NAME property");
+
+            String::from_utf8(title_bytes).map_err(|_| NotSupportedError::new())?
+        };
```

Comment:
Don't we have a `title()` getter already? I'm pretty sure you can just `self.title()`.

Reply:
The `title()` getter returns an empty string.